ofxsMaskMix.h

```diff
@@ -192,16 +192,16 @@ ofxsUnPremult(const PIX *srcPix,
         unpPix[0] = 0.f;
         unpPix[1] = 0.f;
         unpPix[2] = 0.f;
-        unpPix[3] = srcPix[0] / (float)maxValue;
```

Comment:
Yes you have right – unfortunately – it looks like for me – that VS2013 treat this as dividing two integer/byte operator that caused mess in colors. I can check this one again if you think I made somewhere mistake.  

 

From: Frédéric Devernay [mailto:notifications@github.com] 
Sent: Sunday, March 08, 2015 8:47 PM
To: devernay/openfx-supportext
Cc: AnDyXX
Subject: Re: [openfx-supportext] Changes related to use plugins on hosts that use byte/integer color bit depth. (#1)

 

In ofxsMaskMix.h <https://github.com/devernay/openfx-supportext/pull/1#discussion_r26008250> :

> @@ -192,16 +192,16 @@ ofxsUnPremult(const PIX *srcPix,
>          unpPix[0] = 0.f;
>          unpPix[1] = 0.f;
>          unpPix[2] = 0.f;
> -        unpPix[3] = srcPix[0] / (float)maxValue;

I don't see the point in casting the left operand of "/". Type promotion is part of the C language.

—
Reply to this email directly or view it on GitHub <https://github.com/devernay/openfx-supportext/pull/1/files#r26008250> .  <https://github.com/notifications/beacon/AA0pYwKZLGCu_PAb0Uow-v9yBNyNEXd9ks5nzJ7EgaJpZM4DrB-C.gif>